## *********************************************************
##
## File autogenerated for the roboservo_sample package
## by the dynamic_reconfigure package.
## Please do not edit.
##
## ********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'upper': 'DEFAULT', 'lower': 'groups', 'srcline': 246, 'name': 'Default', 'parent': 0, 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'cstate': 'true', 'parentname': 'Default', 'class': 'DEFAULT', 'field': 'default', 'state': True, 'parentclass': '', 'groups': [], 'parameters': [{'srcline': 274, 'description': 'A list of joints name', 'max': 'roboservo_3', 'cconsttype': 'const char * const', 'ctype': 'std::string', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Joint_name', 'edit_method': "{'enum_description': 'A list of nodes name', 'enum': [{'srcline': 10, 'description': 'The 1st joint', 'srcfile': '/home/futaba/catkin_ws/src/roboservo_sample/cfg/roboservo_param.cfg', 'cconsttype': 'const char * const', 'value': 'roboservo_1', 'ctype': 'std::string', 'type': 'str', 'name': '1'}, {'srcline': 11, 'description': 'The 2nd joint', 'srcfile': '/home/futaba/catkin_ws/src/roboservo_sample/cfg/roboservo_param.cfg', 'cconsttype': 'const char * const', 'value': 'roboservo_2', 'ctype': 'std::string', 'type': 'str', 'name': '2'}, {'srcline': 12, 'description': 'The 3rd joint', 'srcfile': '/home/futaba/catkin_ws/src/roboservo_sample/cfg/roboservo_param.cfg', 'cconsttype': 'const char * const', 'value': 'roboservo_3', 'ctype': 'std::string', 'type': 'str', 'name': '3'}]}", 'default': 'roboservo_1', 'level': 0, 'min': 'roboservo_1', 'type': 'str'}, {'srcline': 274, 'description': 'Profile velocity', 'max': 6400, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Profile_velocity', 'edit_method': '', 'default': 1000, 'level': 0, 'min': 1, 'type': 'int'}, {'srcline': 274, 'description': 'Profile acceleration', 'max': 30000, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Profile_acceleration', 'edit_method': '', 'default': 1000, 'level': 0, 'min': 1, 'type': 'int'}, {'srcline': 274, 'description': 'Profile decceleration', 'max': 30000, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Profile_deceleration', 'edit_method': '', 'default': 1000, 'level': 0, 'min': 1, 'type': 'int'}], 'type': '', 'id': 0}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

roboservo_param_1 = 'roboservo_1'
roboservo_param_2 = 'roboservo_2'
roboservo_param_3 = 'roboservo_3'
